apps/spruce/src/pages/task/taskTabs/TaskHistory/CommitDetailsCard/index.tsx

```diff
@@ -74,10 +77,16 @@ const CommitDetailsCard: React.FC<CommitDetailsCardProps> = ({
       dispatchToast.error(`Error restarting task: ${err.message}`),
   });
 
+  const isUnmatchingResult = !!(
+    testFailureSearchTerm &&
+    !testResults.some(({ testFile }) => testFile.match(testFailureSearchTerm))
+  );
+
```

Comment:
I think it would be easier to read if the condition was flipped. And you wouldn't have type errors
```suggestion
  const isMatching =
    !testFailureSearchTerm ||
    testResults.some(({ testFile }) => testFile.match(testFailureSearchTerm));

```